x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/queries/WeightedTokensQueryBuilder.java

```diff
@@ -150,8 +151,14 @@ private boolean shouldKeepToken(
     protected Query doToQuery(SearchExecutionContext context) throws IOException {
         final MappedFieldType ft = context.getFieldType(fieldName);
         if (ft == null) {
-            return new MatchNoDocsQuery("The \"" + getName() + "\" query is against a field that does not exist");
+            throw new ElasticsearchParseException("[" + fieldName + "]" + " is not a mapped field");
```

Comment:
Is this going to cause BWC issues?

Reply:
@kderusso I have checked that elasticsearch-ci/bwc-snapshots tests are passing. Do you suggest to conduct any additional tests here?

Reply:
Let me change this part so that we return the `MatchNoDocs` query to preserve the same behavior.

Reply:
Updated